pkg/operators/ebpf/attach.go

```diff
@@ -145,6 +153,82 @@ func (i *ebpfInstance) attachProgram(gadgetCtx operators.GadgetContext, p *ebpf.
 		return link.AttachLSM(link.LSMOptions{
 			Program: prog,
 		})
+	case ebpf.PerfEvent:
+		perfType := uint32(unix.PERF_TYPE_SOFTWARE)
+		perfConfig := uint64(unix.PERF_COUNT_SW_CPU_CLOCK)
+		perfSampleType := uint64(unix.PERF_SAMPLE_RAW)
+		frequency := uint64(0)
+		name, ok := strings.CutPrefix(p.SectionName, perfEventPrefix)
+		if !ok {
+			return nil, fmt.Errorf("perf_event programs require a name")
+		}
+
+		switch tmp := i.config.GetString("programs." + name + ".perf.type"); tmp {
+		case "":
+			return nil, fmt.Errorf("perf.type not specified for program %q", name)
+		case "software":
+			perfType = unix.PERF_TYPE_SOFTWARE
+		default:
+			return nil, fmt.Errorf("unsupported perf.type %q", tmp)
+		}
+		switch tmp := i.config.GetString("programs." + name + ".perf.config"); tmp {
+		case "":
+			return nil, fmt.Errorf("perf.config not specified for program %q", name)
+		case "count_sw_cpu_clock":
+			perfConfig = unix.PERF_COUNT_SW_CPU_CLOCK
+		default:
+			return nil, fmt.Errorf("unsupported perf.config %q", tmp)
+		}
+		switch tmp := i.config.GetString("programs." + name + ".perf.sampleType"); tmp {
+		case "":
+			return nil, fmt.Errorf("perf.sampleType not specified for program %q", name)
+		case "sample_raw":
+			perfSampleType = unix.PERF_SAMPLE_RAW
+		default:
+			return nil, fmt.Errorf("unsupported perf.sampleType %q", tmp)
+		}
+		if tmpFrequency := i.config.GetString("programs." + name + ".sampler.frequency"); tmpFrequency != "" {
+			var err error
+			frequency, err = strconv.ParseUint(tmpFrequency, 10, 64)
+			if err != nil {
+				return nil, fmt.Errorf("parsing frequency %q for program %q: %w", tmpFrequency, name, err)
+			}
+		} else {
+			return nil, fmt.Errorf("sampler.frequency not specified for program %q", name)
+		}
+		if frequency == 0 {
+			return nil, fmt.Errorf("sampler.frequency is zero for program %q", name)
+		}
+		for cpu := 0; cpu < runtime.NumCPU(); cpu++ {
+			fd, err := unix.PerfEventOpen(
+				&unix.PerfEventAttr{
+					Type:        perfType,
+					Config:      perfConfig,
+					Sample_type: perfSampleType,
+					Sample:      frequency,
+					Bits:        frequencyBit,
```

Comment:
```suggestion
					Bits:        unix.PerfBitFreq,
```

See:
https://pkg.go.dev/golang.org/x/sys/unix
```go
const PerfBitFreq uint64 = CBitFieldMaskBit10
```

Then, we could delete our constant `frequencyBit`.